UiTests/B2CUiTest/B2CUiTest.cs

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Identity;
+using Common;
+using Microsoft.Identity.Lab.Api;
+using Microsoft.Playwright;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.Versioning;
+using System.Threading.Tasks;
+using Xunit;
+using Xunit.Abstractions;
+using TC = Common.TestConstants;
+
+namespace B2CUiTest
+{
+    public class B2CUiTest : IClassFixture<InstallPlaywrightBrowserFixture>
+    {
+        private const string KeyvaultEmailName = "IdWeb-B2C-user";
+        private const string KeyvaultPasswordName = "IdWeb-B2C-password";
+        private const string KeyvaultClientSecretName = "IdWeb-B2C-Client-ClientSecret";
+        private const string NameOfUser = "unknown";
+        private const uint ProcessStartupRetryNum = 3;
+        private const string SampleSolutionFileName = "4-2-B2C-Secured-API.sln";
+        private const uint TodoListClientPort = 5000;
+        private const uint TodoListServicePort = 44332;
+        private const string TraceClassName = "B2C-Login";
+
+        private readonly LocatorAssertionsToBeVisibleOptions _assertVisibleOptions = new() { Timeout = 25000 };
+        private readonly string _sampleClientAppPath;
+        private readonly string _samplePath = Path.Join("4-WebApp-your-API", "4-2-B2C");
+        private readonly string _sampleServiceAppPath;
+        private readonly Uri _keyvaultUri = new("https://webappsapistests.vault.azure.net");
+        private readonly ITestOutputHelper _output;
+        private readonly string _testAssemblyLocation = typeof(B2CUiTest).Assembly.Location;
+
+        public B2CUiTest(ITestOutputHelper output)
+        {
+            _output = output;
+            _sampleClientAppPath = Path.Join(_samplePath, TC.s_todoListClientPath);
+            _sampleServiceAppPath = Path.Join(_samplePath, TC.s_todoListServicePath);
+        }
+
+        [Fact]
+        [SupportedOSPlatform("windows")]
+        public async Task B2C_ValidCreds_LoginLogout()
+        {
+            // Web app and api environmental variable setup.
+            Dictionary<string, Process>? processes = null;
```

Comment:
check the warning, there is one in the GitHub UI

Reply:
enabled nullable in B2C UI test csproj